React/Base/RCTUtils.m

```diff
@@ -885,9 +885,12 @@ BOOL RCTIsLocalAssetURL(NSURL *__nullable imageURL)
       }
 #if !TARGET_OS_OSX // TODO(macOS GH#774)
       image = [UIImage imageWithContentsOfFile:filePath];
-#else // TODO(macOS GH#774)
-      image = [[NSImage alloc] initWithContentsOfFile:filePath]; // TODO(macOS GH#774)
-#endif // TODO(macOS GH#774)
+#else // [TODO(macOS GH#774)
+      // macOS keeps file handles in open state for lifetime of image if "initWithContentsOfFile:" is used with path inside app bundle
+      // Workaround is to load file in data and then convert data to image
+      NSData *data = [NSData dataWithContentsOfFile:filePath];
+      image = [[NSImage alloc] initWithData:data];
```

Comment:
what about https://developer.apple.com/documentation/appkit/nsimage/1519990-initbyreferencingurl? this makes sure it nsimage is lazily loaded.  but isn't the real fix be RN not holding on to nsimage if it is unncessary?

Reply:
I'm gonna vote to keep as is, to keep it closer to the iOS equivalent API. Also, after speaking with Christoph, I see that this change was "stress tested" in their production app, so I'm more partial to it.